components/widgets/settings/bool_setting.tsx

```diff
@@ -1,33 +1,32 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 
-import Setting from './setting.jsx';
+import Setting from './setting';
 
-export default class BoolSetting extends React.Component {
-    static propTypes = {
-        id: PropTypes.string.isRequired,
-        label: PropTypes.node.isRequired,
-        labelClassName: PropTypes.string,
-        helpText: PropTypes.node,
-        placeholder: PropTypes.string.isRequired,
-        value: PropTypes.bool.isRequired,
-        inputClassName: PropTypes.string,
-        onChange: PropTypes.func,
-    };
+type BoolSettingProps = {
```

Comment:
nit: Please simply name it as `Props` (unless there are multiple components in a file).  Same case on other instance.

Reply:
@saturninoabril yes lint test are failing due to prop types not present although prop type is not being used in that file itself.
> This seems to be happening only functional components not for class components
What should be done for that ?